lib/global-styles.php

```diff
@@ -259,58 +259,67 @@ function gutenberg_experimental_global_styles_get_theme() {
  * @return array
  */
 function gutenberg_experimental_global_styles_get_block_data() {
-	// TODO: this data should be taken from the block registry.
-	//
-	// At the moment this array replicates the current capabilities
-	// declared by blocks via __experimentalLineHeight,
-	// __experimentalColor, and __experimentalFontSize.
 	$block_data = array(
-		'global'          => array(
+		'global' => array(
 			'selector' => ':root',
 			'supports' => array(), // By being blank, the 'global' section won't output any style yet.
 		),
-		'core/paragraph'  => array(
-			'selector' => 'p',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h1' => array(
-			'selector' => 'h1',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h2' => array(
-			'selector' => 'h2',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h3' => array(
-			'selector' => 'h3',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h4' => array(
-			'selector' => 'h4',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h5' => array(
-			'selector' => 'h5',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/heading/h6' => array(
-			'selector' => 'h6',
-			'supports' => array( 'line-height', 'font-size', 'color' ),
-		),
-		'core/columns'    => array(
-			'selector' => '.wp-block-columns',
-			'supports' => array( 'color' ),
-		),
-		'core/group'      => array(
-			'selector' => '.wp-block-group',
-			'supports' => array( 'color' ),
-		),
-		'core/media-text' => array(
-			'selector' => '.wp-block-media-text',
-			'supports' => array( 'color' ),
-		),
 	);
 
+	$supported_features = array(
+		'color'       => '__experimentalColor',
+		'line-height' => '__experimentalLineHeight',
+		'font-size'   => '__experimentalFontSize',
+	);
```

Comment:
not sure how important is it for this PR but noting that some of these features are not booleans.

For example for the "color" one, you can enable "colors" and "gradients" if  the support for these is enabled, theme.json should be able to define three things:

```
background: gradient;
backgroundColor: red;
color: green;
```

Reply:
Yeah, I've been reviewing [the existing style properties](https://github.com/WordPress/gutenberg/issues/22700) and also noticed that background (solid or gradients) was missing from the original hard-coded list in global styles. I'll fix it in this PR.

Reply:
Noting also that priority is important here (we can't apply the backgroundColor before the gradient otherwise it's overridden)

Reply:
I'm thinking that priority can work like it works today (and in CSS, so people's mental models are aligned): all declarations are transformed to CSS in the order they appear in the theme.json. You can't have both a solid & a gradient background, so I think this assumption should be fine.